### ISSUE TITLE
`tsc --noEmit` doesn't catch inline default export type errors

### ISSUE THREAD

**Issue by dborysov** [Microsoft/TypeScript#27091]
**TypeScript Version:**  3.1.0-dev.20180914

**Search Terms:**
default export not type-safe
tsc --watch gives more errors
--noEmit doesn't catch error
tsc noemit inline default export

**Code**

```ts
// ourParam.ts:

// not type checked
export default {
  nonExistingParam: '42',
}

// type checked
// const a = {
//   nonExistingParam: '42'
// }
// export default a
```
```ts
// index.ts:

import ourParam from './ourParam'

export interface OurType {
    a: number
  }

const ourMethod = (param: OurType) => {
  console.log(param)
}

const result = ourMethod(ourParam)
```
1. run `npx tsc`
1. 👍 we get the error `Argument of type '{ nonExistingParam: string; }' is not assignable to parameter of type 'OurType'`
1. run `npx tsc --noEmit`
1. 👎 command doesn't give any errors
1. run `npx tsc --noEmit --watch`
1. 👍 we get the error `Argument of type '{ nonExistingParam: string; }' is not assignable to parameter of type 'OurType'`
1. comment out `non type checked` case in `ourParam.ts` file and uncomment the `type checked` case
1. run `npx tsc --noEmit`
1. 👍 we get the error `Argument of type '{ nonExistingParam: string; }' is not assignable to parameter of type 'OurType'`

**Expected behavior:**
* On step 4 we should see the same error as on step 2 and 6
* Two cases in `ourParam.ts` should work the same

**Actual behavior:**
* We don't see any errors on step 4
* Two cases in `ourParam.ts` work differently

**Playground Link:** 
[here](https://stackblitz.com/edit/typescript-sfqmv3)

**Related Issues:** 
-
